pkg/pool-weighted/contracts/lib/WeightCompression.sol

```diff
@@ -22,77 +22,37 @@ import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
-
     using FixedPoint for uint256;
 
-    /**
-     * @dev Convert a 16-bit value to full FixedPoint
-     */
-    function uncompress16(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(FixedPoint.ONE).divUp(type(uint16).max);
-    }
-
-    /**
-     * @dev Compress a FixedPoint value to 16 bits
-     */
-    function compress16(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(type(uint16).max).divUp(FixedPoint.ONE);
+    // If no normalization value is given, assume it is 1.
+    function compress(uint256 value, uint256 bitLength) internal pure returns (uint256) {
+        return compress(value, bitLength, FixedPoint.ONE);
     }
 
-    /**
-     * @dev Convert a 31-bit value to full FixedPoint
-     */
-    function uncompress31(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(FixedPoint.ONE).divUp(_UINT31_MAX);
-    }
-
-    /**
-     * @dev Compress a FixedPoint value to 31 bits
-     */
-    function compress31(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(_UINT31_MAX).divUp(FixedPoint.ONE);
-    }
+    // If a normalization factor is given (e.g., 10_000), normalize against that.
+    function compress(
+        uint256 value,
+        uint256 bitLength,
+        uint256 normalizedTo
+    ) internal pure returns (uint256) {
+        uint256 maxValue = (1 << bitLength) - 1;
 
-    /**
-     * @dev Convert a 32-bit value to full FixedPoint
-     */
-    function uncompress32(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(FixedPoint.ONE).divUp(type(uint32).max);
+        return value.mulUp(maxValue).divUp(normalizedTo);
     }
 
-    /**
-     * @dev Compress a FixedPoint value to 32 bits
-     */
-    function compress32(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(type(uint32).max).divUp(FixedPoint.ONE);
+    // Reverse a compression operation (normalized to 1).
+    function uncompress(uint256 value, uint256 bitLength) internal pure returns (uint256) {
+        return uncompress(value, bitLength, FixedPoint.ONE);
     }
 
-    /**
-     * @dev Convert a 64-bit value to full FixedPoint (assuming a max value of ONE)
-     */
-    function uncompress64(uint256 value) internal pure returns (uint256) {
-        return uncompress64(value, FixedPoint.ONE);
-    }
-
-    /**
-     * @dev Compress a FixedPoint value to 64 bits (assuming a max value of ONE)
-     */
-    function compress64(uint256 value) internal pure returns (uint256) {
-        return compress64(value, FixedPoint.ONE);
-    }
-
-    /**
-     * @dev Convert a 64-bit value to full FixedPoint
-     */
-    function uncompress64(uint256 value, uint256 maxValue) internal pure returns (uint256) {
-        return value.mulUp(maxValue).divUp(type(uint64).max);
-    }
+    // Reverse a compression operation (normalized to a given value).
+    function uncompress(
```

Comment:
Seems like we should rename this be `decompress`?

Reply:
This is like unregister vs deregister. It was kind of a toss-up there, but we decided on deregister.

decompress *is* usually the word used for the opposite of compression (e.g., in other software), so I'm happy to change it, unless there are strong objections.

Just in straight English terms, "uncompressed" is normally an adjective meaning it was never compressed. You can make a verb out of anything, but decompress is the more commonly used verb for reversing a compression operation.

Reply:
For a people divided by a common language, we're agreeing a lot on English lately.

I need to decompress after a long day of PR wrangling!